### PR TITLE
catppuccin-sddm: init unstable-2022-12-03

### PR DESCRIPTION
**Pull request by Chaostheorie**
###### Description of changes

Adds the catppuccin sddm theme. I might need some guidance on where the package should be located. For now it is located as a QT package however that can be changed.

This derivation has been runnning on my setup for the better part of 6 months and I wanted to upstream it. I'm not quite sure how the runtime dependencies on QT quickcontrols 2 and sddm components should be integrated as they appear to be part of sddm anyways.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] ~~x86_64-darwin~~ (sddm not applicable)
  - [ ] ~~aarch64-darwin~~ (sddm not applicable)
- [ ] ~~For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix /manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))~~ (sddm not applicable)
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/top-level/qt5-packages.nix [hunk 1]
```diff
@@ -235,6 +235,7 @@ in (kdeFrameworks // plasmaMobileGear // plasma5 // plasma5.thirdParty // kdeGea
   # Not a library, but we do want it to be built for every qt version there
   # is, to allow users to choose the right build if needed.
   sddm = callPackage ../applications/display-managers/sddm { };
+  sddm-catppuccin = callPackage ../applications/display-managers/sddm/sddm-catppuccin.nix {};
```
Comment (SuperSandro2000) [inline pkgs/top-level/qt5-packages.nix:238]: In qt packages only offficial qt things or libraries are meant to be put in. The theme should just live in top-level.

File: pkgs/applications/display-managers/sddm/sddm-catppuccin.nix [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    repo = "sddm";
+    # Main @ bde6932
+    rev = "bde6932e1ae0f8fdda76eff5c81ea8d3b7d653c0";
+    sha256 = "ceaK/I5lhFz6c+UafQyQVJIzzPxjmsscBgj8130D4dE=";
+  };
+
+  meta = {
+    description = "Soothing pastel theme for SDDM";
+    homepage = "https://github.com/catppuccin/sddm";
+    maintainers = [ ];
```
Comment (SuperSandro2000) [inline pkgs/applications/display-managers/sddm/sddm-catppuccin.nix:23]: Maintainer must be set.

File: pkgs/applications/display-managers/sddm/sddm-catppuccin.nix [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+  dontBuild = true;
+  installPhase = ''
+    mkdir -p $out/share/sddm/themes
+    mv src/catppuccin-${flavor} $out/share/sddm/themes/catppuccin-${flavor}
+  '';
+
+  src = fetchFromGitHub {
+    owner = "catppuccin";
+    repo = "sddm";
+    # Main @ bde6932
```
Comment (SuperSandro2000) [inline pkgs/applications/display-managers/sddm/sddm-catppuccin.nix:15]:
```suggestion
```

File: pkgs/applications/display-managers/sddm/sddm-catppuccin.nix [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+  pname = "sddm-catppuccin-theme";
+  version = "0.0.0b0";
+
+  dontBuild = true;
+  installPhase = ''
+    mkdir -p $out/share/sddm/themes
+    mv src/catppuccin-${flavor} $out/share/sddm/themes/catppuccin-${flavor}
+  '';
+
+  src = fetchFromGitHub {
```
Comment (SuperSandro2000) [inline pkgs/applications/display-managers/sddm/sddm-catppuccin.nix:12]: src should be ordered after version

File: pkgs/applications/display-managers/sddm/sddm-catppuccin.nix [hunk 5]
```diff
@@ -0,0 +1,27 @@
+{ stdenv, lib, fetchFromGitHub, flavor ? "mocha" }:
+stdenv.mkDerivation {
+  pname = "sddm-catppuccin-theme";
+  version = "0.0.0b0";
+
+  dontBuild = true;
```
Comment (SuperSandro2000) [inline pkgs/applications/display-managers/sddm/sddm-catppuccin.nix:6]: Is this actually required? Is stdenv detecting some makefile?

File: pkgs/applications/display-managers/sddm/sddm-catppuccin.nix [hunk 6]
```diff
@@ -0,0 +1,27 @@
+{ stdenv, lib, fetchFromGitHub, flavor ? "mocha" }:
+stdenv.mkDerivation {
+  pname = "sddm-catppuccin-theme";
+  version = "0.0.0b0";
```
Comment (SuperSandro2000) [inline pkgs/applications/display-managers/sddm/sddm-catppuccin.nix:4]:
```suggestion
{ stdenv, lib, fetchFromGitHub, flavor ? "mocha" }:

stdenv.mkDerivation {
  pname = "sddm-catppuccin-theme";
  version = "unstable-XXXX-XX-XX";
```
XXXX-XX-XX should be the date of the commit in iso format

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by Chaostheorie** [comment]
Thank you for the quick review, I will fix the suggested changes asap.

File: pkgs/applications/display-managers/sddm/sddm-catppuccin.nix (hunk 6 above)
Comment (Chaostheorie) [reply pkgs/applications/display-managers/sddm/sddm-catppuccin.nix]: resolved in [517edfb](https://github.com/NixOS/nixpkgs/pull/240990/commits/517edfbaeaad7c040ffa212334e4584816f2a2d1)

File: pkgs/applications/display-managers/sddm/sddm-catppuccin.nix (hunk 5 above)
Comment (Chaostheorie) [reply pkgs/applications/display-managers/sddm/sddm-catppuccin.nix]: this wasn't required. It must've been an artifact from another theme.

File: pkgs/applications/display-managers/sddm/sddm-catppuccin.nix (hunk 4 above)
Comment (Chaostheorie) [reply pkgs/applications/display-managers/sddm/sddm-catppuccin.nix]: resolved in [517edfb](https://github.com/NixOS/nixpkgs/pull/240990/commits/517edfbaeaad7c040ffa212334e4584816f2a2d1)

File: pkgs/top-level/qt5-packages.nix (hunk 1 above)
Comment (Chaostheorie) [reply pkgs/top-level/qt5-packages.nix]: it has been moved to the top-level and renamed to fit in with the other catppuccin themes.

File: pkgs/applications/display-managers/sddm/sddm-catppuccin.nix (hunk 2 above)
Comment (Chaostheorie) [reply pkgs/applications/display-managers/sddm/sddm-catppuccin.nix]: resolved in [517edfb](https://github.com/NixOS/nixpkgs/pull/240990/commits/517edfbaeaad7c040ffa212334e4584816f2a2d1)

File: pkgs/applications/display-managers/sddm/sddm-catppuccin.nix (hunk 3 above)
Comment (Chaostheorie) [reply pkgs/applications/display-managers/sddm/sddm-catppuccin.nix]: resolved in [517edfb](https://github.com/NixOS/nixpkgs/pull/240990/commits/517edfbaeaad7c040ffa212334e4584816f2a2d1)

File: maintainers/maintainer-list.nix [hunk 7]
```diff
@@ -1,61 +1,61 @@
...
-     * test the validity of the format with:
-         nix-build lib/tests/maintainers.nix
+  When editing this file:
+  * keep the list alphabetically sorted, check with:
+  nix-instantiate --eval maintainers/scripts/check-maintainers-sorted.nix
+  * test the validity of the format with:
+  nix-build lib/tests/maintainers.nix
 
-    See `./scripts/check-maintainer-github-handles.sh` for an example on how to work with this data.
+  See `./scripts/check-maintainer-github-handles.sh` for an example on how to work with this data.
```
Comment (SuperSandro2000) [inline maintainers/maintainer-list.nix:58]: Please remove those formatting changes

File: maintainers/maintainer-list.nix [hunk 8]
```diff
@@ -18297,7 +18308,7 @@
   zzzsy = {
     email = "me@zzzsy.top";
     github = "zzzsyyy";
-    githubId = 	59917878;
+    githubId = 59917878;
```
Comment (SuperSandro2000) [inline maintainers/maintainer-list.nix:18311]: we can keep this

File: maintainers/maintainer-list.nix [hunk 9]
```diff
@@ -3127,6 +3127,17 @@
     githubId = 718298;
     name = "Michael Livshin";
   };
+  cobalt = {
+    name = "Cobalt";
+
+    email = "cobalt@cobalt.rocks";
+    github = "Chaostheorie";
+    githubId = 42151227;
+
```
Comment (SuperSandro2000) [inline maintainers/maintainer-list.nix:3136]:
```suggestion
    email = "cobalt@cobalt.rocks";
    github = "Chaostheorie";
    githubId = 42151227;
```

File: pkgs/data/themes/catppuccin-sddm/default.nix [hunk 10]
```diff
@@ -0,0 +1,25 @@
+{ stdenv, lib, fetchFromGitHub, flavor ? "mocha" }:
+stdenv.mkDerivation {
+  pname = "sddm-catppuccin-theme";
+  version = "unstable-2022-12-03";
```
Comment (keenanweaver) [inline pkgs/data/themes/catppuccin-sddm/default.nix:4]:
```suggestion
  version = "unstable-2023-08-08";
```

File: pkgs/data/themes/catppuccin-sddm/default.nix [hunk 11]
```diff
@@ -0,0 +1,25 @@
...
+
+  src = fetchFromGitHub {
+    owner = "catppuccin";
+    repo = "sddm";
+    rev = "bde6932e1ae0f8fdda76eff5c81ea8d3b7d653c0";
+    sha256 = "ceaK/I5lhFz6c+UafQyQVJIzzPxjmsscBgj8130D4dE=";
```
Comment (keenanweaver) [inline pkgs/data/themes/catppuccin-sddm/default.nix:10]:
```suggestion
    rev = "7fc67d1027cdb7f4d833c5d23a8c34a0029b0661";
    hash = "sha256-SjYwyUvvx/ageqVH5MmYmHNRKNvvnF3DYMJ/f2/L+Go=";
```
Bumped to latest

**Comment by Chaostheorie** [comment]
Closed in favor of #255808

**Pull request closed without merging**